src/lib/Components/Inbox/Conversations/Preview/InvoicePreview.tsx

```diff
@@ -99,29 +111,77 @@ const InvoiceStateButton: React.SFC<InvoiceStateButtonProps> = ({ state }) => {
   }
 }
 
-export const InvoicePreview: React.SFC<Props> = ({ invoice, onSelected }) =>
-  <TouchableHighlight onPress={onSelected} underlayColor={colors["gray-light"]}>
-    <Container>
-      <Icon source={require("../../../../../../images/payment_request.png")} />
-      <TextContainer>
-        <PaymentRequest>Payment request</PaymentRequest>
-        <CostLabel>
-          {invoice.total}
-        </CostLabel>
-      </TextContainer>
-      <TextContainer>
-        <InvoiceStateButton state={invoice.state} />
-      </TextContainer>
-    </Container>
-  </TouchableHighlight>
-
-export default createFragmentContainer(
+interface State {
+  hasReceivedNotification: boolean
+  subscription: EmitterSubscription
+}
+
+export class InvoicePreview extends React.Component<Props, State> {
+  componentWillMount() {
+    const { conversationId, relay, invoice } = this.props
+    const { ARNotificationsManager } = NativeModules
+    const notificationsManagerEmitter = new NativeEventEmitter(ARNotificationsManager)
+    const paymentRequestPaidCallback = notification => {
+      if (notification.url === invoice.payment_url) {
+        // Optimistically update the UI, but also refetch.
+        this.setState({
+          hasReceivedNotification: true,
+        })
+        relay.refetch({ conversationId, invoiceId: invoice.lewitt_invoice_id }, null, null, { force: true })
+      }
+    }
+    const subscription = notificationsManagerEmitter.addListener("PaymentRequestPaid", paymentRequestPaidCallback)
+
+    this.setState({
+      subscription,
+    })
+  }
+
+  componentWillUnmount() {
+    if (this.state.subscription) {
+      this.state.subscription.remove()
+    }
+  }
+
+  render() {
+    const { invoice, onSelected } = this.props
+
+    return (
+      <TouchableHighlight onPress={invoice.state === "UNPAID" ? onSelected : null} underlayColor={colors["gray-light"]}>
+        <Container>
+          <Icon source={require("../../../../../../images/payment_request.png")} />
+          <TextContainer>
+            <PaymentRequest>Payment request</PaymentRequest>
+            <CostLabel>
+              {invoice.total}
+            </CostLabel>
+          </TextContainer>
+          <TextContainer>
+            <InvoiceStateButton invoiceState={invoice.state} isOptimisticUpdate={this.state.hasReceivedNotification} />
+          </TextContainer>
+        </Container>
+      </TouchableHighlight>
+    )
+  }
+}
+
+export default createRefetchContainer(
   InvoicePreview,
   graphql`
     fragment InvoicePreview_invoice on Invoice {
       payment_url
       state
       total
+      lewitt_invoice_id
+    }
+  `,
+  graphql`
+    query InvoicePreviewRefetchQuery($conversationId: String!, $invoiceId: String!) {
+      me {
+        invoice(conversationId: $conversationId, invoiceId: $invoiceId) {
```

Comment:
I thought you had added `invoice` as a root field, but in this case I still think using the `node` interface is a better choice.

In this case the `me` resolver probably doesn’t do anything due to the blacklisting we’ve got going on in there, but in general there’s no reason to need to traverse the graph at all if you know the unique ID, ensuring we never accidentally and unnecessarily trigger requests.

Reply:
I did add it as a root field...to `me` (!) 

I see what you're saying about refetching a node given the unique node ID, vs re-specifying the whole schema, which was sort of why I wanted to make it more root-like.

Reply:
I assume you’re kidding, but for posterity I’ll note that `me` itself is a root field, as it lies at the root of the graph that our schema exposes, but fields nested under `me` are not root fields.

Reply:
Got it, I was just following our loose convention that authenticated stuff go within `me`. But that's just sort of how we've done it, its totally not any sort of standard.

It's probably time we depart from that.